process/check.py

```diff
@@ -200,6 +209,14 @@ def psutil_wrapper(self, process, method, accessors, *args, **kwargs):
             self.log.debug("psutil method %s not implemented", method)
         except psutil.AccessDenied:
             self.log.debug("psutil was denied acccess for method %s", method)
+            if method == 'num_fds' and Platform.is_unix():
+                try:
+                    # It is up the agent's packager to grant corresponding sudo policy on unix platforms
+                    result = int(subprocess.check_output(['sudo', sys.executable, __file__, method, str(process.pid)]))
```

Comment:
This is a cool solution but I am not certain it will work with Agent6, where we're calling the python library from within golang. At least for now, you'd have to ensure this happens only in Agent5. You can import the version from here: https://github.com/DataDog/dd-agent/blob/master/config.py#L44.

@masci how do you think we would be able to do this in Agent6?

Reply:
As a second note here, why did you choose to execute this file rather than including that as a string that would be passed to the interpreter? I think that's more likely to work in Agent6 than this is.

Reply:
Hi @gmmeyer, I chose to invoke this script on purpose for two reasons:

1) For security reasons: passing the process id to this script requires wildcards. Knowing that this script can only accept a process id and not anything else feels safe to me. 
    Passing an arbitrary string to an interpreter with a wildcard has dangerous implications: https://blog.compass-security.com/2012/10/dangerous-sudoers-entries-part-4-wildcards/
    An alternative may be to have several sudoers entries to match all possible pid lengths:
    ```
dd-agent ALL=NOPASSWD: /opt/datadog-agent/embedded/bin/python -c "import psutil; import sys; psutil.num_fds(sys.argv[1])" [1-9]
dd-agent ALL=NOPASSWD: /opt/datadog-agent/embedded/bin/python -c "import psutil; import sys; psutil.num_fds(sys.argv[1])" [1-9][0-9]
dd-agent ALL=NOPASSWD: /opt/datadog-agent/embedded/bin/python -c "import psutil; import sys; psutil.num_fds(sys.argv[1])" [1-9][0-9][0-9]
dd-agent ALL=NOPASSWD: /opt/datadog-agent/embedded/bin/python -c "import psutil; import sys; psutil.num_fds(sys.argv[1])" [1-9][0-9][0-9][0-9]
dd-agent ALL=NOPASSWD: /opt/datadog-agent/embedded/bin/python -c "import psutil; import sys; psutil.num_fds(sys.argv[1])" [1-6][0-5][0-9][0-9][0-9]
    ```
    
2) For easy distribution: as the omnibus packaging rename files, this solution should work everywhere, no need to know the path to the script.

Reply:
Okay, so that makes sense. However, in Agent v6 we're not using a python interpreter, though, it's embedded within a go application. I think (I haven't verified this) that this would just call the go application instead of the python interpreter.

Reply:
I tested github.com/sbinet/go-python that is used in agent 6 to wrap the python interpreter and it seems to be ok:

`main.go`:
```
package main

import (
        "fmt"
        "log"

        "github.com/sbinet/go-python"
)

func init() {
        err := python.Initialize()
        if err != nil {
                log.Panic(err.Error())
        }
}

func main() {
        module := python.PyImport_ImportModule("testsys")
        if module == nil {
                log.Fatal("could not import 'testsys'")
        }

        name := module.GetAttrString("__name__")
        if name == nil {
                log.Fatal("could not getattr '__name__'")
        }
        defer name.DecRef()
        fmt.Printf("testsys.__name__: %q\n", python.PyString_AsString(name))

        testsys := module.GetAttrString("testsys")
        if testsys == nil {
                log.Fatal("could not getattr 'testsys'")
        }
        defer testsys.DecRef()

        o1 := testsys.CallFunction()
        if o1 == nil {
                log.Fatal("could not call 'testsys.testsys()'")
        }
        fmt.Printf("%s\n", python.PyString_AsString(o1))
        o1.DecRef()
}
```

`testsys.py`:
```
#!/usr/bin/env python2

import sys

def testsys():
    return "sys.executable=" + sys.executable
```

Result:
```
# PYTHONPATH=.:$PYTHONPATH go run ./main.go 
testsys.__name__: "testsys"
sys.executable=/usr/bin/python
```

Reply:
I'm down to merge this then! I'll approve it!

Reply:
Can you rebase it and then I'll approve, sorry for taking so long on this 😄

Reply:
No prob, rebased!

However, I did not yet implement the part about adding an option to the process check's configuration to enable the use of sudo, disabled by default.